core/transaction_pool/impl/pool_moderator_impl.hpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_POOL_MODERATOR_IMPL_HPP
+#define KAGOME_POOL_MODERATOR_IMPL_HPP
+
+#include <map>
+
+#include "transaction_pool/impl/clock_impl.hpp"
+#include "transaction_pool/pool_moderator.hpp"
+
+namespace kagome::transaction_pool {
+
+  class PoolModeratorImpl : public PoolModerator {
+    static constexpr size_t kDefaultExpectedSize = 2048;
+
+    using SystemClock = Clock<std::chrono::system_clock>;
+    using Map = std::map<common::Hash256, SystemClock::TimePoint>;
+
+   public:
+    /**
+     * @param ban_for amount of time for which a transaction is banned
+     * @param clock a clock used to determine when it is time to unban a
+     * transaction
+     * @param expected_size expected maximum number of banned transactions. If
+     * significantly exceeded, some transactions will be removed from ban list
+     */
+    explicit PoolModeratorImpl(
+        SystemClock::Duration ban_for = std::chrono::minutes(30),
+        std::shared_ptr<SystemClock> clock =
+            std::make_shared<kagome::transaction_pool::SystemClock>(),
```

Comment:
better remove this default. this line couples `pool_moderator_impl` to `SystemClock`, which is against DI principles

Reply:
I had doubts if it's a good idea, either, but I came to the conclusion that as long as SystemClock is a part of TransactionPool implementation alike PoolModeratorImpl (so they're sort of in the same scope anyway), and this default value proves very useful (because I do suspect that default clock will be changed only in tests), it is conceptually okay to make it this way.